include/evmc/hex.hpp

```diff
@@ -113,4 +113,24 @@ inline std::optional<bytes> from_hex(std::string_view hex)
         return {};
     return bs;
 }
+
+/// Decodes hex-encoded string into custom type T with .bytes array of uint8_t.
+///
+/// The decoded bytes are right aligned in case the input is smaller than the result type.
```

Comment:
Why?

Reply:
Because they imitate BE numbers.

Reply:
Yes, but left/right does not have clear meaning.

Reply:
> When the input is smaller than the result type, the result is padded with zeros on the left (the result bytes of lowest indices are filled with zeros).

?

Reply:
Yes, an option can be added if needed, but so far this is only used for literals and loading storage values from JSON.

Reply:
Perhaps add `// TODO: support left alignment too with a flag`.